src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -1495,9 +1490,7 @@ private static unsafe int ApplyMask(Span<byte> toMask, int mask, int maskIndex)
 
                 if (toMaskEnd - toMaskPtr >= sizeof(int))
                 {
-                    int rolledMask = BitConverter.IsLittleEndian ?
-                        (int)BitOperations.RotateRight((uint)mask, maskIndex * 8) :
-                        (int)BitOperations.RotateLeft((uint)mask, maskIndex * 8);
+                    int rolledMask = (int)BitOperations.RotateRight((uint)mask, maskIndex * 8);
```

Comment:
I was thinking of using `TensorPrimitives.Xor` here for the `sizeof(int)` elements but I think we can't guarantee alignment of the `toMask` for `MemoryMarshal.Cast` here..